playground/src/helpers/localStorage/bookmarkedEndpoints.ts

```diff
@@ -0,0 +1,25 @@
+const BOOKMARK_LOCALSTORAGE_KEY = "bookmarked_endpoints";
+
+export function persistEndpointBookmarkStatus(name: string, status: boolean) {
+	const currentBookmaked = getLocalStorageBookmarks();
+	if (status) {
+		// bookmark
+		setLocalStorageBookmarks([...currentBookmaked, name]);
+	} else {
+		// unbookmark
+		setLocalStorageBookmarks(currentBookmaked.filter(n => n !== name));
+	}
+}
+
+export function getLocalStorageBookmarks(): string[] {
+	const content = localStorage.getItem(BOOKMARK_LOCALSTORAGE_KEY);
+	if (content) {
+		return JSON.parse(content);
+	} else {
+		return [];
+	}
+}
+
+export function setLocalStorageBookmarks(names: string[]) {
+	localStorage.setItem(BOOKMARK_LOCALSTORAGE_KEY, JSON.stringify(names));
+}
```

Comment:
This will have some issues with browsers limiting local storage, such as lack of space of Safari's icognito mode. Better check that localStorage is actually working and guard it with a try catch, falling back to memory: https://michalzalecki.com/why-using-localStorage-directly-is-a-bad-idea/

Reply:
solved